### ISSUE TITLE
Mostly non-functional with Sony ILCE-7C

### ISSUE THREAD

**Issue by BenLand100** [gphoto/libgphoto2#946]
I use this camera for astrophotography, and ultimately ended up here after Kstars/Ekos identified a "lack of exposure widget" for this camera as being a gphoto2 error. After debugging with gphoto2 CLI a bit, it appears my (?) ILCE-7C doesn't have any significant functionality, despite being listed as supported by gphoto2.  Any idea how to get to a more capable state?

I assume this is not an issue with my camera, as the same cables+camera work fine under a Windows ecosystem also using Sony's PTP https://github.com/dougforpres/ASCOMSonyCameraDriver

Things to note: 
* Camera is in PC Remote mode (same as windows)
* -PC- emblem shows up on camera indicating it's connected (same as windows)
* Certain things do seem to partially work
* Mostly, core functionality with this camera is missing

Basic capture functionality errors out, camera does nothing:

```
benland100@blandtoo ~ $ gphoto2 --capture-image
ERROR: Could not capture image.                                                
ERROR: Could not capture.
```
Curiously, `--trigger-capture` works, as in the camera takes a picture successfully, but the command itself fails:

```
benland100@blandtoo ~ $ gphoto2 --trigger-capture
ERROR: Could not trigger capture.                                              
*** Error (-1: 'Unspecified error') ***  
```

There are not many config options showing, though I can adjust e.g. `whitebalance` with `--set-config` and  `--get-config` and these seem to take effect on the camera.

```
benland100@blandtoo ~ $ gphoto2 --list-all-config
/main/actions/movie                                                            
Label: Movie Capture
Readonly: 0
Type: TOGGLE
Current: 2
END
/main/actions/opcode
Label: PTP Opcode
Readonly: 0
Type: TEXT
Current: 0x1001,0xparam1,0xparam2
END
/main/status/serialnumber
Label: Serial Number
Readonly: 1
Type: TEXT
Current: 00000000000000003285211106169435
END
/main/status/manufacturer
Label: Camera Manufacturer
Readonly: 1
Type: TEXT
Current: Sony Corporation
END
/main/status/cameramodel
Label: Camera Model
Readonly: 1
Type: TEXT
Current: ILCE-7C
END
/main/status/deviceversion
Label: Device Version
Readonly: 1
Type: TEXT
Current: 2.00
END
/main/status/vendorextension
Label: Vendor Extension
Readonly: 1
Type: TEXT
Current: Sony PTP Extensions
END
/main/imgsettings/whitebalance
Label: WhiteBalance
Readonly: 0
Type: RADIO
Current: Daylight
Choice: 0 Automatic
Choice: 1 Daylight
Choice: 2 Shade
Choice: 3 Cloudy
Choice: 4 Tungsten
Choice: 5 Fluorescent: Warm White
Choice: 6 Fluorescent: Cold White
Choice: 7 Fluorescent: Day White
Choice: 8 Fluorescent: Daylight
Choice: 9 Flash
Choice: 10 Underwater: Auto
Choice: 11 Choose Color Temperature
Choice: 12 Preset 1
Choice: 13 Preset 2
Choice: 14 Preset 3
END
/main/other/5005
Label: White Balance
Readonly: 0
Type: MENU
Current: 4
Choice: 0 2
Choice: 1 4
Choice: 2 32785
Choice: 3 32784
Choice: 4 6
Choice: 5 32769
Choice: 6 32770
Choice: 7 32771
Choice: 8 32772
Choice: 9 7
Choice: 10 32816
Choice: 11 32786
Choice: 12 32800
Choice: 13 32801
Choice: 14 32802
END
```

And the summary is bleak to put it lightly:

```
benland100@blandtoo ~ $ gphoto2 --summary
Camera summary:                                                                
Manufacturer: Sony Corporation
Model: ILCE-7C
  Version: 2.00
  Serial Number: 00000000000000003285211106169435
Vendor Extension ID: 0x11 (1.0)
Vendor Extension Description: Sony PTP Extensions

Capture Formats: 
Display Formats: Association/Directory, JPEG, ARW, MPEG, Unknown(b982)

Device Capabilities:
	File Download, No File Deletion, File Upload
	No Image Capture, No Open Capture, Sony Capture

Storage Devices Summary:

Device Property Summary:
White Balance(0x5005):(readwrite) (type=0x4) Enumeration [2,4,32785,32784,6,32769,32770,32771,32772,7,32816,32786,32800,32801,32802] value: Daylight (4)
F-Number(0x5007): error 2002 on query.
Focus Mode(0x500a): error 2002 on query.
Exposure Metering Mode(0x500b): error 2002 on query.
Flash Mode(0x500c): error 2002 on query.
Exposure Program Mode(0x500e): error 2002 on query.
Exposure Bias Compensation(0x5010): error 2002 on query.
Still Capture Mode(0x5013): error 2002 on query.
DOC Compensation(0xd200): error 2002 on query.
DRangeOptimize(0xd201): error 2002 on query.
Image size(0xd203): error 2002 on query.
Shutter speed(0xd20d): error 2002 on query.
Property 0xd20e: error 2002 on query.
Color temperature(0xd20f): error 2002 on query.
CC Filter(0xd210): error 2002 on query.
Aspect Ratio(0xd211): error 2002 on query.
Focus status(0xd213): error 2002 on query.
Zoom(0xd214): error 2002 on query.
Objects in memory(0xd215): error 2002 on query.
Property 0xd217: error 2002 on query.
Battery Level(0xd218): error 2002 on query.
Picture Effect(0xd21b): error 2002 on query.
AB Filter(0xd21c): error 2002 on query.
Property 0xd21d: error 2002 on query.
ISO(0xd21e): error 2002 on query.
Property 0xd21f: error 2002 on query.
Property 0xd221: error 2002 on query.
Capture Target(0xd222): error 2002 on query.
Property 0xd223: error 2002 on query.
Property 0xd22a: error 2002 on query.
Focus Area(0xd22c): error 2002 on query.
Live View Setting Effect(0xd231): error 2002 on query.
Property 0xd235: error 2002 on query.
Property 0xd236: error 2002 on query.
Property 0xd23d: error 2002 on query.
Property 0xd23e: error 2002 on query.
Property 0xd23f: error 2002 on query.
Property 0xd240: error 2002 on query.
Property 0xd241: error 2002 on query.
Property 0xd242: error 2002 on query.
Property 0xd243: error 2002 on query.
Property 0xd244: error 2002 on query.
Property 0xd245: error 2002 on query.
Property 0xd246: error 2002 on query.
Property 0xd247: error 2002 on query.
Property 0xd248: error 2002 on query.
Property 0xd249: error 2002 on query.
Property 0xd24a: error 2002 on query.
Property 0xd24c: error 2002 on query.
Property 0xd24e: error 2002 on query.
Property 0xd24f: error 2002 on query.
Property 0xd250: error 2002 on query.
Property 0xd251: error 2002 on query.
Property 0xd252: error 2002 on query.
Property 0xd253: error 2002 on query.
Property 0xd254: error 2002 on query.
Property 0xd255: error 2002 on query.
Property 0xd259: error 2002 on query.
Property 0xd25a: error 2002 on query.
Property 0xd25b: error 2002 on query.
Property 0xd25c: error 2002 on query.
Property 0xd25d: error 2002 on query.
Property 0xd25f: error 2002 on query.
Property 0xd260: error 2002 on query.
Property 0xd261: error 2002 on query.
Property 0xd262: error 2002 on query.
Property 0xd263: error 2002 on query.
Property 0xd264: error 2002 on query.
Property 0xd267: error 2002 on query.
Property 0xd268: error 2002 on query.
Property 0xd269: error 2002 on query.
Property 0xd26a: error 2002 on query.
Property 0xd26b: error 2002 on query.
Property 0xd26c: error 2002 on query.
Property 0xd26d: error 2002 on query.
Property 0xd26e: error 2002 on query.
Property 0xd26f: error 2002 on query.
Property 0xd270: error 2002 on query.
Property 0xd271: error 2002 on query.
Property 0xd272: error 2002 on query.
Property 0xd273: error 2002 on query.
Property 0xd274: error 2002 on query.
Property 0xd275: error 2002 on query.
Property 0xd276: error 2002 on query.
Property 0xd278: error 2002 on query.
Property 0xd279: error 2002 on query.
Property 0xd27a: error 2002 on query.
Property 0xd27b: error 2002 on query.
Property 0xd27c: error 2002 on query.
Property 0xd27d: error 2002 on query.
Property 0xd27e: error 2002 on query.
Property 0xd27f: error 2002 on query.
Property 0xd280: error 2002 on query.
Property 0xd283: error 2002 on query.
Property 0xd284: error 2002 on query.
Property 0xd285: error 2002 on query.
Property 0xd295: error 2002 on query.
Autofocus(0xd2c1): error 2002 on query.
Capture(0xd2c2): error 2002 on query.
Property 0xd2c3: error 2002 on query.
Still Image(0xd2c7): error 2002 on query.
Movie(0xd2c8): error 2002 on query.
Property 0xd2c9: error 2002 on query.
Property 0xd2cd: error 2002 on query.
Property 0xd2ce: error 2002 on query.
Property 0xd2cf: error 2002 on query.
Property 0xd2d0: error 2002 on query.
Near Far(0xd2d1): error 2002 on query.
Property 0xd2d2: error 2002 on query.
Property 0xd2d5: error 2002 on query.
Property 0xd2d6: error 2002 on query.
Property 0xd2d7: error 2002 on query.
Property 0xd2d8: error 2002 on query.
Property 0xd2d9: error 2002 on query.
Property 0xd2da: error 2002 on query.
Property 0xd2db: error 2002 on query.
AF Area Position(0xd2dc): error 2002 on query.
Property 0xd2dd: error 2002 on query.
Property 0xd2df: error 2002 on query.
Property 0xd2e0: error 2002 on query.
Property 0xd2e1: error 2002 on query.
Property 0xd2e2: error 2002 on query.
Property 0xd2e4: error 2002 on query.
Property 0xd2e5: error 2002 on query.
Property 0xd2e6: error 2002 on query.
```

**Name the camera**

Sony ILCE-7C (Control)

**libgphoto2 and gphoto2 version**

```
gphoto2 2.5.28

Copyright (c) 2000-2021 Marcus Meissner and others

gphoto2 comes with NO WARRANTY, to the extent permitted by law. You may
redistribute copies of gphoto2 under the terms of the GNU General Public
License. For more information about these matters, see the files named COPYING.

This version of gphoto2 is using the following software versions and options:
gphoto2         2.5.28         x86_64-pc-linux-gnu-gcc, popt(m), exif, cdk, aa, jpeg, readline
libgphoto2      2.5.31         standard camlibs, x86_64-pc-linux-gnu-gcc, no ltdl, EXIF
libgphoto2_port 0.12.2         iolibs: disk ptpip serial usb1 usbdiskdirect usbscsi, x86_64-pc-linux-gnu-gcc, no ltdl, EXIF, USB, serial lockdev lockin
```

**Comment by BenLand100** [comment]
Looks like the fix for #937, which is already in the repo, resolves this, it's just not in the most recent release yet.

**Comment by msmeissn** [comment]
yes, will be in next release
